### ISSUE TITLE
Allow users to select Piped/Invidious instances to play YouTube videos

### ISSUE THREAD

**Issue by candroid-man** [futo-org/grayjay-android#149]
Piped and Invidious are great ways of watching YouTube anonymously, and it is totally decentralized (ran by individual users) which alligns with the goals and morals behind Grayjay. It would also be great way of syncing subscription, history, etc to desktop via Piped/Invidious account.

Admittedly, Grayjay may already use Invidious similarly to NewPipe, however I wasn't able to find anything in your FAQ or website about it.

**Comment by Robbylikes** [comment]
Using invidious to pull videos from YouTube is safe and smart. I had thought the same thing as we all notice YouTube is some super big evil corp.

**Comment by koen-futo** [comment]
This is technically already possible, anybody can make a plugin to do this. https://gitlab.futo.org/videostreaming/grayjay/-/blob/master/plugin-development.md see this how to get started.

**Comment by koen-futo** [comment]
We might make an official plugin at some point, but it is not high on the priority list.

**Comment by candroid-man** [comment]
It would be great if Piped or Invidious was the default option while not logged in to a YouTube account since these projects are more trusted in terms of privacy, thoughts?

**Comment by koen-futo** [comment]
If somebody wants to make an Invidious plugin, we will be glad to help (you can join FUTO chat with questions) and include it in the app in the future.

**Comment by b-risk** [comment]
> If somebody wants to make an Invidious plugin, we will be glad to help (you can join FUTO chat with questions) and include it in the app in the future.


I could be misunderstanding but it feels like this should just be included in the plugin as an option to route through? Because invidious just serves exactly the same content and info as YT, I understand this might be a lower priority for FUTO themselves to implement but if someone were to make a pull request for the sotck plugin theoretically would FUTO accept it?

**Comment by arn0nym0us** [comment]
> > If somebody wants to make an Invidious plugin, we will be glad to help (you can join FUTO chat with questions) and include it in the app in the future.
> 
> I could be misunderstanding but it feels like this should just be included in the plugin as an option to route through? Because invidious just serves exactly the same content and info as YT, I understand this might be a lower priority for FUTO themselves to implement but if someone were to make a pull request for the stock plugin theoretically would FUTO accept it?

It's not just about routing requests through Piped or Invidious.
A huge benefit would also come from being able to sync Subscriptions, Playlist and watch history to Piped and Invidous.
That would make using Grayjay on multiple Devices a much better experience, cause currently you need to manualy import and export your data between devices.

**Comment by Gamer750** [comment]
I would also like to see this

**Comment by Gamer750** [comment]
Any update

**Comment by Zvonimir-FUTO** [comment]
No update on this that I know of. As Koen already said, if somebody wants to make an Invidious plugin, we will be glad to help (you can join FUTO chat with questions).

**Comment by b-risk** [comment]
> > > If somebody wants to make an Invidious plugin, we will be glad to help (you can join FUTO chat with questions) and include it in the app in the future.
> > 
> > 
> > I could be misunderstanding but it feels like this should just be included in the plugin as an option to route through? Because invidious just serves exactly the same content and info as YT, I understand this might be a lower priority for FUTO themselves to implement but if someone were to make a pull request for the stock plugin theoretically would FUTO accept it?
> 
> It's not just about routing requests through Piped or Invidious. A huge benefit would also come from being able to sync Subscriptions, Playlist and watch history to Piped and Invidous. That would make using Grayjay on multiple Devices a much better experience, cause currently you need to manualy import and export your data between devices.

I think this should be considered as a proxy method for the official YouTube plugin, now that Grayjay sync is a thing the only usecase for something like this would be to proxy videos, as YouTube is ramping up their detection more people are being blocked on an IP level and you can see examples of that with people in GitHub issues, it also has the benefit of being more private as proxying hides your own IP from YT. I'm not sure if Invidious has something like this, but I know [Cobalt](cobalt.tools) has an option to route videos through their server instead. This is still low hanging fruit no doubt as I think a proxy would be the last line of the defense aside from VPN and account login fallback.

**Comment by Gamer750** [comment]
Is anyone working on this
